### PR TITLE
[SPARK-9447] [ML] [PYTHON] Added HasRawPredictionCol, HasProbabilityCol to RandomForestClassifier

### PR DESCRIPTION
**Pull request by jkbradley**
Added HasRawPredictionCol, HasProbabilityCol to RandomForestClassifier, plus doc tests for those columns.

CC: @holdenk @yanboliang

### CODE REVIEW

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -320,8 +322,13 @@ class RandomForestClassifier(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPred
...
     True
     >>> test0 = sqlContext.createDataFrame([(Vectors.dense(-1.0),)], ["features"])
-    >>> model.transform(test0).head().prediction
+    >>> result = model.transform(test0).head()
+    >>> result.prediction
     0.0
+    >>> numpy.argmax(result.probability)
+    0
+    >>> numpy.argmax(result.rawPrediction)
+    0
```
Comment (yanboliang) [inline python/pyspark/ml/classification.py]: Here users may want to know the raw prediction and probability prediction result as the confidence measure, so I think output ```result.probability``` or ```numpy.amax(result.probability)``` is better to demonstrate how to use this feature.

Comment (jkbradley) [inline python/pyspark/ml/classification.py]: That turned out to be hard to test.  Rounding issues made the values slightly off.

Comment (yanboliang) [inline python/pyspark/ml/classification.py]: That's OK.

**Comment by jkbradley** [comment]
OK I'll merge this with master and branch-1.5.  Thanks for looking at it!

**Pull request closed without merging**